### ISSUE TITLE
amp-story: Best practice: add focusable controls in small-screen view

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32504]
# Best practice: add focusable controls in small-screen view

## Pages/screens/components affected
* [Navigation](https://www.gstatic.com/amphtml/stamp/qa/navigation.html)
* [RTL](https://www.gstatic.com/amphtml/stamp/qa/rtl.html)
* [Video](https://www.gstatic.com/amphtml/stamp/qa/video.html)
* [Audio - amp-story background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-story.html)
* [Audio - amp-story-page background audio](https://www.gstatic.com/amphtml/stamp/qa/audio-page.html)
* [Audio - amp-story using amp-audio](https://www.gstatic.com/amphtml/stamp/qa/amp-audio.html)
* [Animation](https://www.gstatic.com/amphtml/stamp/qa/animation.html)
* [Sharing](https://www.gstatic.com/amphtml/stamp/qa/sharing.html)
* [Tooltip](https://www.gstatic.com/amphtml/stamp/qa/tooltip.html)
* [Page attachment](https://www.gstatic.com/amphtml/stamp/qa/page-attachment.html)
* [Branching](https://www.gstatic.com/amphtml/stamp/qa/branching.html#page=page2)
* [Consent](https://www.gstatic.com/amphtml/stamp/qa/consent.html)
* [Prerender - Image](https://www.gstatic.com/amphtml/stamp/qa/prerender-img.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [Prerender - Video](https://www.gstatic.com/amphtml/stamp/qa/prerender-video.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [AMP Story Player - Player](https://www.gstatic.com/amphtml/stamp/qa/player.html)
* [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html)

## Description
Good design means supporting different modes of interaction, because there are many reasons why someone might use a keyboard instead of a mouse, and if keyboard interaction is not fully supported the website is effectively unusable.

In small-screen/smartphone mode, stories don't present users with any visible previous/next/replay controls. Keyboard users can still navigate using cursor keys - therefore nominally passing the broad requirements for [WCAG 2.1 Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard). However, this may create usability issues for users of keyboard-like interfaces, who may not be able to easily trigger these key presses. It is also arguably not immediately obvious that keyboard users *can* in fact use cursor keys to navigate.

## User impact
Someone with mobility problems will often use a keyboard instead of a mouse to navigate a web page. A screen reader user will also tend to use a keyboard if they cannot see the screen to use the mouse. Therefore if the website does not properly support keyboard interaction a large number of people are effectively prevented from using it.

In this case, while nominally AMP stories in small-screen/smartphone view *are* keyboard-operable using cursor keys to move between pages, this may not be obvious to users. This may also cause issues for users of keyboard-like interfaces (such as switch access users), as there is no obvious focusable control that can be navigated to and activated to navigate between pages.

It's worth emphasising that the small-screen/smartphone view of AMP Stories won't only be used on an actual touch screen operated smartphone. This view will also be used by desktop users that have sized/zoomed their browser window to a small-enough size to trigger the "smartphone" view, but are still navigating using keyboard or mouse. And even on actual smartphones, users may be navigating with an external mouse or keyboard. For this reason it's crucial that this view also work well for keyboard users.

## Required solution
We recommend programmatically detecting if a user is navigating with a keyboard (for instance, using a helper library such as [What Input?](https://github.com/ten1seven/what-input)) and showing visible navigation controls that can be focused and activated using the keyboard - similar to the controls that appear in [Tooltip](https://www.gstatic.com/amphtml/stamp/qa/tooltip.html) when the embedded tweet is clicked/tapped.

![Visible previous/next controls in the tooltip sample](https://user-images.githubusercontent.com/54933393/107247692-7390b980-6a29-11eb-8fbf-009be8be475e.png)

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Open the AMP Story in small-screen/smartphone view
2. Check that, when navigating through stories using a keyboard, visible navigation controls (previous/next) appear
3. Check that these controls can be focused using `Tab` and activated using `Space` and `Enter`

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG21/#keyboard)

## More information
* [WebAIM: Keyboard](https://webaim.org/techniques/keyboard/)

## Test data
Test date: January 2021

**Comment by processprocess** [comment]
`amp-mode-keyboard-active` is added to the `body` element. 
We can display visible buttons when this class is added.

<img width="461" alt="Screen Shot 2021-02-24 at 4 18 30 PM" src="https://user-images.githubusercontent.com/3860311/109067017-f7020a00-76bb-11eb-9cb8-a3f68fed5495.png">

**Comment by mszylkowski** [comment]
We're already adding buttons for navigation, I think we'd be ok by adding some visual hint that these buttons are for navigation (eg: background-image of a chevron) when the class `amp-mode-keyboard-active` is present on the page.

Current buttons are transparent but focusable on mobile, and they are properly labeled for screen readers:
```html
<button class="i-amphtml-story-button-move" aria-label="Next page"></button>
```
